### PR TITLE
Set minSDK to 21 as compose as

### LINKED ISSUE #1: Support Android API < 28
Now in Android API 28

image

### PR DESCRIPTION
**Pull request by adamglin0**

### CODE REVIEW

File: compose-shadow/build.gradle.kts [hunk 1]
```diff
@@ -73,7 +73,6 @@ android {
 
     defaultConfig {
         minSdk = libs.versions.androidMinSdk.get().toInt()
-        lint.targetSdk = libs.versions.androidTargetSdk.get().toInt()
```
Comment (skaldebane) [inline compose-shadow/build.gradle.kts]: The target SDK shouldn't be removed.

However, `lint.targetSdk` is an error, it only affects the linter. I fixed many instances of this in another PR, guess this one went unnoticed :D

This should be
```kotlin
targetSdk = libs.versions.andtoidTargetSdk.get().toInt()
```

**Review by skaldebane — CHANGES_REQUESTED** [review]

File: compose-shadow/build.gradle.kts (hunk 1 above)
Comment (adamglin0) [reply compose-shadow/build.gradle.kts]: Yes, with your help, the project has been standardized a lot💗. I will modify this.

Comment (adamglin0) [reply compose-shadow/build.gradle.kts]: > The target SDK shouldn't be removed.
> 
> However, `lint.targetSdk` is an error, it only affects the linter. I fixed many instances of this in another PR, guess this one went unnoticed :D
> 
> This should be
> 
> ```kotlin
> targetSdk = libs.versions.andtoidTargetSdk.get().toInt()
> ```

```
'var targetSdk: Int?' is deprecated. Will be removed from library DSL in v9.0. Use testOptions. targetSdk or/ and lint. targetSdk instead.
```

I will restore lint.targetSdk.

Comment (skaldebane) [reply compose-shadow/build.gradle.kts]: Glad I was of help! It's been fun to work on this, my first involvement in an OSS project as a collaborator 😁

**Review by skaldebane — COMMENTED** [review]
LGTM!
One last thing though, was this tested on an unsupported device? Just to make sure that it doesn't crash.

**Comment by adamglin0** [comment]
> LGTM!
> One last thing though, was this tested on an unsupported device? Just to make sure that it doesn't crash.

Just make sure no crash. The feature that support android21 is working in #43.

**Review by citytexi — APPROVED** [review]
LGTM!!!

Just curious - what's the reason for setting min to 21?

**Review by skaldebane — APPROVED** [review]

**Comment by skaldebane** [comment]
@citytexi It's to support every version Compose itself supports. This is a UI library for Compose, so it kinda makes sense.

While the default KMP templates sets the minSdk to SDK 24 by default, most of `androidx` supports SDK 21 just fine (some even lower), and there's not much UI-wise that is different on Android 5 compared to, say, Android 7, so it's not a maintenance hurdle if you're already trying to support Android 7.

It might make sense for an app, but a library should always strive to maximize compatibility, so apps get to make the choice. Even as a no-op.

**Pull request merged**